src/contracts/tokens/RewardPool.sol

```diff
@@ -0,0 +1,237 @@
+    uint256 public rewardPerTokenStored;
+    /// @inheritdoc IRewardPool
+    uint256 public periodFinish = 0;
+    /// @inheritdoc IRewardPool
+    uint256 public rewardRate = 0;
+    /// @inheritdoc IRewardPool
+    uint256 public lastUpdateTime;
+    /// @inheritdoc IRewardPool
+    uint256 public queuedRewards = 0;
+    /// @inheritdoc IRewardPool
+    uint256 public currentRewards = 0;
+    /// @inheritdoc IRewardPool
+    uint256 public historicalRewards = 0;
+    /// @inheritdoc IRewardPool
+    uint256 public rewardPerTokenPaid = 0;
+    /// @inheritdoc IRewardPool
+    uint256 public rewards = 0;
+
+    // --- Init ---
+
+    /**
+     * @param _rewardToken Address of the reward token
+     */
+    constructor(address _rewardToken) {
+        if (_rewardToken == address(0)) revert RewardPool_InvalidRewardToken();
+        rewardToken = IERC20(_rewardToken);
+        _params.duration = 7 days;
+        _params.newRewardRatio = 830;
+    }
+
+    // --- Methods ---
+
+    /// @inheritdoc IRewardPool
+    function lastTimeRewardApplicable()
+        public
+        view
+        returns (uint256 _lastTime)
+    {
+        return Math.min(block.timestamp, periodFinish);
+    }
+
+    /// @inheritdoc IRewardPool
+    function totalStaked() public view returns (uint256) {
+        return _totalStaked;
+    }
+
+    /// @inheritdoc IRewardPool
+    function stake(
+        uint256 _wad
+    ) external updateReward(msg.sender) isAuthorized {
+        if (_wad == 0) revert RewardPool_StakeNullAmount();
+        _totalStaked += _wad;
+        emit RewardPool_Staked(msg.sender, _wad);
+    }
+
+    /// @inheritdoc IRewardPool
+    function increaseStake(uint256 _wad) external isAuthorized {
+        if (_wad == 0) revert RewardPool_IncreaseStakeNullAmount();
+        _totalStaked += _wad;
+        emit RewardPool_IncreaseStake(msg.sender, _wad);
+    }
+
+    /// @inheritdoc IRewardPool
+    function decreaseStake(uint256 _wad) external isAuthorized {
+        if (_wad == 0) revert RewardPool_DecreaseStakeNullAmount();
+        if (_wad > _totalStaked) revert RewardPool_InsufficientBalance();
+        _totalStaked -= _wad;
+        emit RewardPool_DecreaseStake(msg.sender, _wad);
+    }
+
+    /// @inheritdoc IRewardPool
+    function withdraw(
+        uint256 _wad,
+        bool _claim
+    ) external updateReward(msg.sender) isAuthorized {
+        if (_wad == 0) revert RewardPool_WithdrawNullAmount();
+        if (_wad > _totalStaked) revert RewardPool_InsufficientBalance();
+        _totalStaked -= _wad;
+        emit RewardPool_Withdrawn(msg.sender, _wad);
+        if (_claim) {
+            _getReward(msg.sender);
+        }
+    }
+
+    /// @inheritdoc IRewardPool
+    function getReward() external updateReward(msg.sender) isAuthorized {
+        _getReward(msg.sender);
+    }
+
+    function _getReward(address _account) internal {
+        uint256 _reward = earned(_account);
+        if (_reward > 0) {
+            rewards = 0;
```

Comment:
is this meant to be only an admin function? the total rewards should not be set to 0 when sending an individual their claim. I dont think this is correct
`rewards` here is a gloabl var

Reply:
So since this is only called by StakingManager, the global `rewards` var is the amount of unclaimed rewards the StakingManager has earned for that pool. So when it is claimed it goes back to 0 as the StakingManager has withdrawn all the available rewards.

Reply:
In that case theres no need to pass along `account`. Its going to the StakingManager every time.